### PR TITLE
feat(4): added handler for create room modal button

### PR DESCRIPTION
**Pull request by Davis856**
- Added handler for creating a room. Now the room is created and you're redirected into the respective room.
- Messages do not work for now - they have to be solved in another feature (this is due to adapter)

### CODE REVIEW

File: lib/chat_app_web/live/home_live.ex [hunk 1]
```diff
@@ -37,4 +42,35 @@ defmodule ChatAppWeb.HomeLive do
...
+          "password" => password
+        },
+        socket
+      ) do
+    is_private =
+      if is_private == "on" do
+        true
+      else
+        false
+      end
```
Comment (sutgeorge) [inline lib/chat_app_web/live/home_live.ex]: I know this is not incredibly important, however I think this if statement could be replaced with a simple case statement, something like this:
```elixir
is_private = 
  case is_private do 
    "on" -> true
    _ -> false
  end
```

But really, I also used an if statement if I'm not mistaken, so it's not essential to remove it.

File: lib/chat_app_web/live/home_live.ex [hunk 2]
```diff
@@ -37,4 +42,35 @@ defmodule ChatAppWeb.HomeLive do
...
+
+    room = %{
+      "room_name" => room_name,
+      "owner_name" => owner_name,
+      "max_participants" => 20,
```
Comment (sutgeorge) [inline lib/chat_app_web/live/home_live.ex]: Is there a particular reason why the room has a capacity of 20 participants? Perhaps that number should be chosen in the modal, although I think it is ok for now to just have it 20 by default.

File: lib/chat_app_web/live/home_live.ex [hunk 3]
```diff
@@ -37,4 +42,35 @@ defmodule ChatAppWeb.HomeLive do
...
+      "is_private" => is_private,
+      "password" => password,
```
Comment (sutgeorge) [inline lib/chat_app_web/live/home_live.ex:63]: Maybe we could hash room passwords later on 😝

File: lib/chat_app_web/live/home_live.ex (hunk 1 above)
Comment (Davis856) [reply lib/chat_app_web/live/home_live.ex]: I can try with that as well, the syntax is weird enough as it is

File: lib/chat_app_web/live/home_live.ex (hunk 3 above)
Comment (Davis856) [reply lib/chat_app_web/live/home_live.ex:63]: Yeah, but for now I just wanted to have it working, will add as an idea

File: lib/chat_app_web/live/home_live.ex (hunk 2 above)
Comment (Davis856) [reply lib/chat_app_web/live/home_live.ex]: Hardcoded, talked at daily and we said that unless we have something to do with this field (which we will, once we do the handling for joining a room as well), we hardcode it

Comment (salajandan) [reply lib/chat_app_web/live/home_live.ex]: @EndlessRecurrence i agree.
We should be able to set the maximum number of participants as long as it is greater than 2.
But we should have a default value of, let's say, 20.

Comment (Davis856) [reply lib/chat_app_web/live/home_live.ex]: > @EndlessRecurrence i agree.
> We should be able to set the maximum number of participants as long as it is greater than 2.
> But we should have a default value of, let's say, 20.

Alrighty then, will modify

File: lib/chat_app/contexts/rooms.ex [hunk 4]
```diff
@@ -2,7 +2,7 @@ defmodule ChatApp.Contexts.Rooms do
   alias ChatApp.Structs.Room
   alias ChatApp.Repo
 
-  def list_rooms(), do: Repo.all(Room) |> Repo.preload([:messages])
+  def list_rooms(), do: Repo.all(Room) |> Repo.preload([:messages]) |> Repo.preload([:users])
```
Comment (salajandan) [inline lib/chat_app/contexts/rooms.ex]: Why not preload both `:messages` and `:users` in the same call?

File: lib/chat_app_web/live/home_live.ex [hunk 5]
```diff
@@ -37,4 +42,35 @@ defmodule ChatAppWeb.HomeLive do
...
+          "is_private" => is_private,
+          "password" => password
+        },
+        socket
+      ) do
+    is_private =
+      if is_private == "on" do
+        true
+      else
+        false
```
Comment (salajandan) [inline lib/chat_app_web/live/home_live.ex]: No need for that much ceremony.
```elixir
is_private_room = is_private == "on"
```

**Review by salajandan — CHANGES_REQUESTED** [review]

File: lib/chat_app_web/live/home_live.ex (hunk 5 above)
Comment (Davis856) [reply lib/chat_app_web/live/home_live.ex]: thank god, i was sure this was not the proper way to do it but I couldn't find a better way

File: lib/chat_app/contexts/rooms.ex (hunk 4 above)
Comment (Davis856) [reply lib/chat_app/contexts/rooms.ex]: Didn't realise for some reason that the argument is a list.

modified

**Review by salajandan — APPROVED** [review]

**Pull request merged**